src/js/main.js

```diff
@@ -0,0 +1,81 @@
+export const DiscourseRADParser = () => {
+  const radTabsElements = document.querySelectorAll('.rad-tabs-element');
+
+  radTabsElements.forEach((radTabsElement) => {
+    setDefaults(radTabsElement);
+    setupDropdowns(radTabsElement);
+  });
+
+  toggleElements();
+}
+
+const setDefaults = radTabsElement => {
+  const dropdowns = radTabsElement.querySelectorAll('select');
+
+  dropdowns.forEach((dropdown) => {
+    const name = dropdown.name;
+
+    const options = dropdown.querySelectorAll('option');
+    let defaultWasSet = false;
+    options.forEach((option) => {
+      if (localStorage.getItem(name) === option.value) {
+        defaultWasSet = true;
+        option.setAttribute('selected', 'selected');
+      }
+    });
+
+    if (!defaultWasSet) {
+      const firstOption = dropdown.firstElementChild;
+      firstOption.setAttribute('selected', 'selected');
+
+      if (localStorage.getItem(name) === null) {
+        localStorage.setItem(name, firstOption.value);
```

Comment:
This feels strange. You are actively recording a choice the user did not make. Therefore if the user moves page and the next page has the options in a different order the localstorage will select it randomly.

Reply:
I was scratching my head how to set a default of a rad criteria. And I came up with the idea of setting the default as the very first option of each unique dropdown name. 

If the user moves page. I do check to see if the `localStorge` is set already and if it is, then I just use the existing option. It's never random.